### PR TITLE
Add kind/proposal label to Cluster API

### PR DESCRIPTION
**Pull request by vincepri**
Signed-off-by: Vince Prignano <vincepri@vmware.com>

/assign @cblecker @ncdc

### CODE REVIEW

File: label_sync/labels.yaml [hunk 1]
```diff
@@ -1227,6 +1227,12 @@ repos:
         addedBy: anyone
   kubernetes-sigs/cluster-api:
     labels:
+      - color: e11d21
```
[the remark this answers is not in the document]
Comment (vincepri) [reply label_sync/labels.yaml]: copied the wrong one

[the remark this answers is not in the document]
Comment (vincepri) [reply label_sync/labels.yaml]: fixed, good catch

File: label_sync/labels.yaml [hunk 2]
```diff
@@ -1227,6 +1227,12 @@ repos:
         addedBy: anyone
   kubernetes-sigs/cluster-api:
     labels:
+      - color: c7def8
+        description: Categorizes issue or PR as related to a proposal.
```
Comment (ncdc) [inline label_sync/labels.yaml]:
```suggestion
        description: Issues or PRs related to proposals
```

Comment (vincepri) [reply label_sync/labels.yaml]: I can change it, although the other kind/* labels have this style of description

**Pull request merged**